iterative/utils/helpers.go

```diff
@@ -89,16 +91,36 @@ func SetId(d *schema.ResourceData) {
 	}
 }
 
-func LoadGCPCredentials() string {
-	credentialsData := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS_DATA")
-	if len(credentialsData) == 0 {
-		credentialsPath := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
-		if len(credentialsPath) > 0 {
-			jsonData, _ := os.ReadFile(credentialsPath)
-			credentialsData = string(jsonData)
+func MultiEnvLoadFirst(envs []string) string {
+	for _, val := range envs {
+		if env_value := os.Getenv(val); env_value != "" {
+			return env_value
 		}
 	}
-	return credentialsData
+	return ""
+}
+
+func GCPCoerceOIDCCredentials(rawCreds []byte) (string, error) {
+	hack := make(map[string]interface{})
+	err := json.Unmarshal(rawCreds, &hack)
+	if err != nil {
+		return "", err
+	}
+	saString := fmt.Sprint(hack["service_account_impersonation_url"])
+	// saString example: "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/cml@cml-pulse.iam.gserviceaccount.com:generateAccessToken"
+	if saString == "" {
+		return "", errors.New("[GCP OIDC] Unable to load service_account_impersonation_url")
+	}
+	// from saString, yank this
+	// -----------------------v.........v----------------------
+	// (...serviceAccounts/cml@cml-pulse.iam.gserviceaccount.com:...)
+	atIndex := strings.Index(saString, "@") + 1
+	iamIndex := strings.Index(saString, ".iam.")
+	if atIndex == -1 || iamIndex == -1 {
+		return "", errors.New("[GCP OIDC] Failed to get Project ID from service_account_impersonation_url")
+	}
+	projectID := saString[atIndex:iamIndex]
+	return projectID, nil
```

Comment:
What about having [100 problems](https://xkcd.com/1171/)? 😄 
```suggestion
func GCPCoerceOIDCCredentials(credentialsJSON []byte) (string, error) {
	credentials := make(map[string]interface{})

	if err := json.Unmarshal(credentialsJSON, &credentials); err != nil {
		return "", err
	}

	if url, ok := credentials["service_account_impersonation_url"].(string); ok {
		re := regexp.MustCompile("^https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/.+?@(?P<project>.+).iam.gserviceaccount.com:generateAccessToken$")
		if match := re.FindStringSubmatch(url); match != nil {
			return match[1], nil
		}
		return "", errors.New("failed to get project identifier from service_account_impersonation_url")
	}

	return "", errors.New("unable to load service_account_impersonation_url")
```

Reply:
I do tend to avoid it. What you have here is still pretty readable and tests well. 
/tomorrow I will apply this and run my sanity checks to be safe but I like it.